libbeat/common/kubernetes/util.go

```diff
@@ -30,10 +30,29 @@ import (
 	"k8s.io/client-go/tools/clientcmd"
 	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 
+	"github.com/pkg/errors"
+
 	"github.com/elastic/beats/v7/libbeat/logp"
 )
 
-const defaultNode = "localhost"
+const namespaceFilePath = "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
+
+type HostDiscoveryUtils interface {
+	GetNamespace() (string, error)
+	GetPodName() (string, error)
+	GetMachineID() string
+}
+
+// DiscoverKubernetesNodeOpts includes options for discovering kubernetes node
+type DiscoverKubernetesNodeOpts struct {
+	ConfigHost  string
+	Client      kubernetes.Interface
+	IsInCluster bool
+	HostUtils   HostDiscoveryUtils
+}
+
+// Hostdiscoveryutils implements functions of HostDiscoveryUtils interface
+type Hostdiscoveryutils struct{}
```

Comment:
Nit. Use camel case.
```suggestion
type HostDiscoveryUtils struct{}
```

Or as this is an specific implementation of an interface with the same name, use a name related to the implementation, or just `DefaultDiscoveryUtils`.